From 1a48d839dc138730bfa07a2af2a05c9995e06e50 Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Tue, 13 Nov 2018 14:05:56 +0200
Subject: [PATCH 5112/5242] MLK-20189 sound: asoc: fix unsigned compared
 against 0

commit  8c94563a3f6ef2e18ee828ae7aa2196d1d7ca3f7 from
https://source.codeaurora.org/external/imx/linux-imx.git

An unsigned value can never be negative, so this test will always
evaluate the same way.

Fixes Coverity issue CID3618200

Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_micfil.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_micfil.c b/sound/soc/fsl/fsl_micfil.c
index 72bffea..18f97d9 100644
--- a/sound/soc/fsl/fsl_micfil.c
+++ b/sound/soc/fsl/fsl_micfil.c
@@ -2136,7 +2136,7 @@ static ssize_t micfil_hwvad_handler(struct kobject *kobj,
 	if (ret < 0)
 		return -EINVAL;
 
-	if (vad_channel >= 0 && vad_channel <= 7) {
+	if (vad_channel <= 7) {
 		micfil->vad_channel = vad_channel;
 		ret = enable_hwvad(dev, true);
 		if (ret) {
-- 
1.7.9.5

